### ISSUE TITLE
TF+XLA+Mixed Precision: Keras fails to compute gradients

### ISSUE THREAD

**Issue by itmo153277** [keras-team/keras#20931]
Keras fails to compute gradients for autoencoder-esce model using Tensorflow backend with mixed precision and jit compilation enabled.
See code here: [colab](https://colab.research.google.com/gist/itmo153277/6f38f43d724e173a30115d12f2a40e07/upsamplerelugrad.ipynb).

This is caused by `UpSampling2D` layer. When gradients are computed, the type is resolved as `float32` instead of `float16`, and this causes Relu that comes next to throw a dtype mismatch exception.

The only working workaround I found is explicitly setting dtype to `float32` for `UpSampling2D` layer. This inserts a `cast` node inbetween `relu` and `upsample` which helps dealing with dtype conversion.

Not sure which project this issue should be submitted to: Keras, TF or XLA

**Comment by divyashreepathihalli** [comment]
Looks like `tf.image.resize(..., method='bilinear')` gradient op produces float 32 tensors
Keras casts forward pass output to float16
consequent layers produce `float16` gradients - leading to XLA error
We would need to re-implement tf.image.resize(..., method='bilinear`) or explicitly set dtype to float32 in tf backend
Works fine on other backends
A workaround for now is to cast it to float 32 like you have. We will try to add a fix.

**Comment by itmo153277** [comment]
Ping to avoid stale status

**Comment by jeffcarp** [comment]
This issue is caused by upstream TensorFlow/XLA gradient tensor casting in `tf.image.resize` rather than Keras core (and works as expected on JAX and PyTorch backends). The recommended workaround is to explicitly specify `dtype="float32"` on the `UpSampling2D` layer. Closing as an upstream TF issue.
